pipeline/builder/builder.go

```diff
@@ -168,6 +187,99 @@ func (b *Builder) buildRunJobStage(index int, s config.Stage) (*types.RunJobStag
 	return rjs, nil
 }
 
+func (b *Builder) buildV2ManifestStage(index int, s config.Stage) (*types.ManifestStage, error) {
+	ds := &types.ManifestStage{
+		StageMetadata:           buildStageMetadata(s, "deployManifest", index, b.isLinear),
+		Account:                 s.Account,
+		CloudProvider:           "kubernetes",
+		Location:                "",
+		ManifestArtifactAccount: "embedded-artifact",
+		ManifestName:            "",
+		Moniker: types.Moniker{
+			App:     b.pipeline.Application,
+			Cluster: fmt.Sprintf("%s-%s", b.pipeline.Application, s.Account),
+			Detail:  "",
+			Stack:   "",
+		},
+		Relationships: types.Relationships{LoadBalancers: []interface{}{}, SecurityGroups: []interface{}{}},
+		Source:        "text",
+	}
+
+	parser := NewManfifestParser(b.pipeline, b.basePath)
+
+	for _, group := range s.Deploy.Groups {
+		manifest, err := parser.ManifestFromScaffold(group)
+
+		if err != nil {
+			return nil, err
+		}
+
+		switch t := manifest.(type) {
+		case *v1beta1.Deployment:
+			_, err = buildDeployment(t, group)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		j, err := json.Marshal(manifest)
+
+		if err != nil {
+			return nil, err
+		}
+		ds.Manifests = append(ds.Manifests, json.RawMessage(j))
+	}
+
+	return ds, nil
+}
+
+func (b *Builder) buildV2RunJobStage(index int, s config.Stage) (*types.ManifestStage, error) {
+	ds := &types.ManifestStage{
+		StageMetadata:           buildStageMetadata(s, "deployManifest", index, b.isLinear),
+		Account:                 s.Account,
+		CloudProvider:           "kubernetes",
+		Location:                "",
+		ManifestArtifactAccount: "embedded-artifact",
+		ManifestName:            "",
+		Moniker: types.Moniker{
+			App:     b.pipeline.Application,
+			Cluster: fmt.Sprintf("%s-%s", b.pipeline.Application, s.Account),
+			Detail:  "",
+			Stack:   "",
+		},
+		Relationships: types.Relationships{LoadBalancers: []interface{}{}, SecurityGroups: []interface{}{}},
+		Source:        "text",
+	}
+
+	parser := NewManfifestParser(b.pipeline, b.basePath)
+	obj, err := parser.ManifestFromScaffold(s.RunJob)
+
+	if err != nil {
+		return nil, err
+	}
+
+	switch t := obj.(type) {
+	case *corev1.Pod:
+		if s.RunJob.Container != nil {
+			t.Spec.Containers[0].Command = s.RunJob.Container.Command
+			t.Spec.Containers[0].Args = s.RunJob.Container.Args
+		}
+	case *v1beta1.Deployment:
+		return nil, ErrDeploymentJob
```

Comment:
Why not just make this the default case? Technically we could use _any_ manifest type and get out of this switch statement if its not a pod or deployment.

Reply:
yep, amending this